.github/workflows/pull.yml

```diff
@@ -734,66 +750,82 @@ jobs:
 
           echo "Tests complete."
   test-mps-dtype:
-    uses: pytorch/test-infra/.github/workflows/macos_job.yml@main
```

Comment:
This test is very flaky because it fails whenever a machine with python 3.9 is allocated.  Change to always setup python 3.10.